apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/Trace/CloudTrace.cs

```diff
@@ -119,13 +120,20 @@ private CloudTrace(string projectId, TraceConfiguration config = null, TraceServ
         /// <summary>
         /// Initialize tracing for this application.
         /// </summary>
-        /// <param name="projectId">The Google Cloud Platform project ID.</param>
         /// <param name="application">The Http application.</param>
+        /// <param name="projectId">Optional if running on Google App Engine or Google Compute Engine.
+        ///     The Google Cloud Platform project ID. If running on GAE or GCE the project ID will be
+        ///     detected from the platform.</param>
         /// <param name="config">Optional trace configuration, if unset the default will be used.</param>
         /// <param name="client">Optional trace client, if unset the default will be used.</param>
-        public static void Initialize(string projectId, HttpApplication application, TraceConfiguration config = null, TraceServiceClient client = null)
+        public static void Initialize(HttpApplication application, string projectId = null, TraceConfiguration config = null, TraceServiceClient client = null)
         {
             GaxPreconditions.CheckNotNull(application, nameof(application));
+
+            // Trace does not use monitored resources, only detect the monitored resource if
+            // a project id is needed.
+            projectId = projectId ?? CommonUtils.GetAndCheckProjectId(
+                projectId, MonitoredResourceBuilder.FromPlatform());
```

Comment:
Is `MonitoredResourceBuilder` accessible from CommunUtils? If so, make the second parameter optional and defaulted to null and have `GetAndCheckProjectId` call `MonitoredResourceBuilder.FromPlatform()` if necessary. Then these kinds of call sites can just look like this:
```csharp
projectId = CommonUtils.GetAndCheckProjectId(projectId);
```